precompiles/bank/method_deposit.go

```diff
@@ -105,36 +89,6 @@ func (c *Contract) deposit(
 		}
 	}
 
-	// Check for enough bank's allowance.
-	// function allowance(address owner, address spender) public view virtual override returns (uint256)
-	resAllowance, err := c.CallContract(
-		ctx,
-		&c.fungibleKeeper,
-		c.zrc20ABI,
-		zrc20Addr,
-		"allowance",
-		[]interface{}{caller, ContractAddress},
-	)
-	if err != nil {
-		return nil, &ptypes.ErrUnexpected{
-			When: "allowance",
-			Got:  err.Error(),
-		}
-	}
-
-	allowance, ok := resAllowance[0].(*big.Int)
-	if !ok {
-		return nil, &ptypes.ErrUnexpected{
-			Got: "ZRC20 allowance returned an unexpected type",
-		}
-	}
-
-	if allowance.Cmp(amount) < 0 || allowance.Cmp(big.NewInt(0)) <= 0 {
-		return nil, &ptypes.ErrInvalidAmount{
-			Got: allowance.String(),
-		}
-	}
-
 	// The process of creating a new cosmos coin is:
 	// - Generate the new coin denom using ZRC20 address,
 	//   this way we map ZRC20 addresses to cosmos denoms "zevm/0x12345".
```

Comment:
// - Mint coins.
	// - Send coins to the caller.
	
	maybe we should mention that the coins are minted to fungible module , and transferred from fungible to caller

Reply:
Changed in f20129df6881a8c92d6fc46e4ef56479823a2177